File: processor/src/main/java/com/github/snowdream/toybricks/processor/ProcessingException.java
package com.github.snowdream.toybricks.processor;

import javax.lang.model.element.Element;

/**
 * Created by snowdream on 17/2/12.
 */
public class ProcessingException extends Exception {

  Element element;

  public ProcessingException(Element element, String msg, Object... args) {
    super(String.format(msg, args));
    this.element = element;
  }

  public Element getElement() {
    return element;
  }
}

File: processor/src/main/kotlin/com/github/snowdream/toybricks/processor/ProcessingException.kt
package com.github.snowdream.toybricks.processor

import javax.lang.model.element.Element

/**
 * Created by snowdream on 17/2/12.
 */
class ProcessingException(element: Element, msg: String, vararg args: String?) : Exception(String.format(msg, *args)) {

    var element: Element
        internal set

    init {
        this.element = element
    }
}
